### ISSUE TITLE
what is `self` in the class based view?

### ISSUE THREAD

**Issue by badbye** [aio-libs/aiohttp#2719]
## Long story short

What is `self` in the class-based view?

```
class CheckUser(web.View):
    @routes.get('/check')
    async def get(self):
        print('type of self: ', type(self))  
```

It should print the `CheckUser` class, however, it prints a `Request` object.
`type of self:  <class 'aiohttp.web_request.Request'>`

**Comment by asvetlov** [comment]
Everything works as expected:
```
from aiohttp import web

class View(web.View):
    async def get(self):
        return web.Response(text=str(type(self)))


app = aiohttp.web.Application()
app.router.add_view('/', View)

web.run_app(app)
```
```
curl http://localhost:8080/
<class '__main__.View'>
```

**Comment by badbye** [comment]
@asvetlov 
Would you mind to try again by using the decorator function?
```
import aiohttp
from aiohttp import web

routes = web.RouteTableDef()


class CheckUser(web.View):
    @routes.get('/')
    async def get(self):
        return web.Response(text=str(type(self)) + ' aiohttp: ' + aiohttp.__version__)


app = web.Application()
app.router.add_routes(routes)


web.run_app(app)
```

```
curl http://127.0.0.1:8080
<class 'aiohttp.web_request.Request'> aiohttp: 3.0.0b4
```

**Comment by asvetlov** [comment]
Aha, I see.
You should use `@router.view` decorator for this. `@router.get` assumes that a decorated function is a plain web handler, not a web view.

The decorator available in aiohttp 3.0 only.
The version will be released tomorrow but you can install beta package form here: https://pypi.python.org/pypi/aiohttp/3.0.0b4

**Comment by badbye** [comment]
Good to know, thanks. Looking forward the 3.0 version.
